azurerm/resource_arm_cognitive_account.go

```diff
@@ -209,14 +220,33 @@ func resourceArmCognitiveAccountRead(d *schema.ResourceData, meta interface{}) e
 		d.Set("location", azureRMNormalizeLocation(*location))
 	}
 
-	if err := d.Set("sku", flattenCognitiveAccountSku(resp.Sku)); err != nil {
+	if err = d.Set("sku", flattenCognitiveAccountSku(resp.Sku)); err != nil {
 		return fmt.Errorf("Error setting `sku`: %+v", err)
 	}
 
 	if props := resp.AccountProperties; props != nil {
 		d.Set("endpoint", props.Endpoint)
 	}
 
+	keys, err := client.ListKeys(ctx, resourceGroup, name)
+
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] Not able to obtain keys for Cognitive Services Account %q in Resource Group %q - removing from state!", name, resourceGroup)
```

Comment:
Yeah, I wasn't sure about this. I was thinking that the resource can't be used without the keys so fail hard. But actually because it is a 2 step process to read the resource and then the keys it probably makes more sense to return the error